test/integration/scheduler/queue_test.go

```diff
@@ -512,6 +518,34 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods:          sets.New("pod2"),
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name:         "Pod rejected with PVC by the CSI plugin is requeued when the pod with the same required PVC is deleted",
```

Comment:
I think it would be also good to test the other scenarios as well: adding a CSI node and adding a PVC. You can also add a one more pod that won't be requeued when the event happen to check the filtering.